crates/sparrow-backend/src/pipeline_schedule.rs

```diff
@@ -49,29 +49,42 @@ pub fn pipeline_schedule(steps: &IndexSlice<StepId, [Step]>) -> Vec<Pipeline> {
 }
 
 /// Return true if the step is "pipeline breaking".
-fn is_pipeline_breaker(index: StepId, step: &Step, references: &IndexVec<StepId, usize>) -> bool {
+///
+/// A "pipeline breaker" must start a new pipeline.
+fn is_pipeline_breaker(
```

Comment:
Changed the scheduling a bit. Now, a sequence of transforms will become a pipeline. If one of them has multiple consumers, then it will end the pipeline.